### DISCUSSION TITLE
driver.getContexts() not returning any contexts for iFrames

### DISCUSSION THREAD

**Discussion by RyanRedpath** [webdriverio/webdriverio#6457] [Q&A]
Hi,

I am writing tests for a mobile app, and as part of a testing user journey I need to interact with an iFrame, so I am hoping that switching context to a webview will work for this. However, in the past past if I switched context by using the code below I would be able to view contexts, and get something like "Contexts: NATIVE_APP,WEBVIEW_XXXX" which is great, and I can switch to  WEBVIEW_XXXX. 

let contexts = driver.getContexts();
console.log(contexts); 
driver.switchContext(contexts[1]);

However, when I try this on with switching to an iFrame I get this back: "[Function] { skip: [Function], only: [Function] }".

Anyone have any experience on switching to iFrames? Or know if it's possible?!

Thanks,
Ryan

**Comment by christian-bromann** [comment]
@RyanRedpath not sure what you are "getting back" but a little code snippet would be helpful here.

**Reply by RyanRedpath** [reply]
@christian-bromann  thanks for your reply, but I think I have found the resolution to the switchContext. 

Whether I can interact with the iFrame is another matter. Once I try `driver.switchToFrame($("#wp-cl-wp-div-iframe"));` I receive the error "An unknown server-side error occurred while processing the command. Original error: Blocked a frame with origin "null" from accessing a cross-origin frame. Protocols, domains, and ports must match." - looks to be related to CORS? Is there any way around this do you know?

**Reply by christian-bromann** [reply]
> Is there any way around this do you know?

No idea
